app/models/concerns/episode_media.rb

```diff
@@ -131,19 +226,48 @@ def media_ready?(must_be_complete = true)
   end
 
   def media_status
-    states = media.map(&:status).uniq
+    states = if override?
+      [external_media_resource&.status]
+    else
+      media.map(&:status).uniq
+    end
     if !(%w[started created processing retrying] & states).empty?
       "processing"
     elsif states.any? { |s| s == "error" }
       "error"
     elsif states.any? { |s| s == "invalid" }
       "invalid"
-    elsif media_ready?
+    elsif media_ready? || override_ready?
       "complete"
     end
   end
 
   def media_url
     media.first.try(:href)
   end
+
+  def override_ready?
+    override? && external_media_ready?
+  end
+
+  def override_processing?
+    override? && !external_media_ready?
+  end
+
+  def override?
+    medium_override? || !enclosure_override_url.blank?
+  end
+
+  def external_media_ready?
+    external_media_resource&.status_complete?
+  end
+
+  def analyze_external_media
```

Comment:
Feels like this could be wrapped into the existing `copy_media` (which yeah, is a misnomer... really means any async process for media).  That would keep things more consistent, around when async jobs get fired.  Thus far, we always do that in `copy_media` called from usually the controller.  Rather than activerecord hooks.